src/Compilers/CSharp/Portable/Symbols/AssemblySymbol.cs

```diff
@@ -440,7 +440,24 @@ internal bool RuntimeSupportsStaticAbstractMembersInInterfaces
             get => RuntimeSupportsFeature(SpecialMember.System_Runtime_CompilerServices_RuntimeFeature__VirtualStaticsInInterfaces);
         }
 
-        private bool RuntimeSupportsFeature(SpecialMember feature)
+        /// <summary>
+        /// Whether the target runtime supports numeric IntPtr types.
+        /// This test hook should be removed once TargetFramework.Net70 is added.
+        /// Tracked by https://github.com/dotnet/roslyn/issues/61235
+        /// </summary>
+        internal virtual bool RuntimeSupportsNumericIntPtr
+        {
+            get
+            {
+                // CorLibrary should never be null, but that invariant is broken in some cases for MissingAssemblySymbol.
+                // Tracked by https://github.com/dotnet/roslyn/issues/61262
+                return CorLibrary?.RuntimeSupportsNumericIntPtr == true;
```

Comment:
>RuntimeSupportsNumericIntPtr [](http://example.com/codeflow?start=35&length=28)

Could this access and the one below go into an infinite loop when the current assembly is the CorLibrary?

Reply:
I don't think so. If `CorLibrary` is available, then we'll call `MetadataOrSourceAssemblySymbol.RuntimeSupportsNumericIntPtr`. If it's missing then we'll call `MissingCorLibrarySymbol.RuntimeSupportsNumericIntPtr`.

FWIW, I followed the same test hook pattern from DIM: [https://github.com/dotnet/roslyn/pull/17927](https://github.com/dotnet/roslyn/pull/17927)